tools/expotools/src/commands/AndroidBuildPackages.ts

```diff
@@ -328,9 +350,23 @@ async function action(options: ActionOptions) {
     }
   }
 
-  await _updateExpoViewAsync(packages.filter(pkg => packagesToBuild.includes(pkg.name)), options.sdkVersion);
+  try {
+    await _updateExpoViewAsync(packages.filter(pkg => packagesToBuild.includes(pkg.name)), options.sdkVersion);
+  } catch (e) {
+    await _exitHandler();
+    throw e;
+  }
 }
 
+async function _exitHandler(): Promise<void> {
+  if (Object.keys(savedFiles).length) {
+    console.log('Exited early, cleaning up...');
+    await _restoreFilesAsync();
+  }
+}
+
+process.on('SIGINT', _exitHandler);
```

Comment:
do you want to handle other signals as well (SIGKILL, SIGTERM) or just SIGINT?

Reply:
sure, shouldn't hurt anything to handle those cases as well